### PR TITLE
Implement new output variable Surface Internal User Specified Location Temperature for Construction:InternalSource

### PR DESCRIPTION
**Pull request by RKStrand**
## Pull request overview

For radiant systems that use a Construction:InternalSource, EnergyPlus has a feature that allows the user to report a temperature internal to the construction.  This is useful for a variety of reasons and also for non-radiant systems potentially.  The problem was that though the construction definition made it possible for the user to request this temperature the code to use this information properly in the code was lacking.  This defect fix corrects this problem and now EnergyPlus properly produces output that shows the internal temperature.  This issue was logged as GitHub Issue #3478.
### Work Checklist

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
- [ x ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
- [ x ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
  - Defect: This pull request repairs GitHub Issue #3478
### Review Checklist

This will not be exhaustively relevant to every PR.
- [x] Code style (parentheses padding, variable names)
- [x] Functional code review (it has to work!)
- [x] If defect, results of running current develop vs this branch should exhibit the fix
- [x] CI status: all green or justified
- [x] Performance: CI Linux results include performance check -- verify this
- [x] Unit Test(s)
- C++ checks:
  - [x] Argument types
  - [x] If any virtual classes, ensure virtual destructor included, other things
- n/a IDD changes:
- [x] If new idf included, locally check the err file and other outputs
- [x] Documentation changes in place
- [x] Changed docs build successfully

### CODE REVIEW

**Comment by mjwitte** [comment]
@Myoldmopar Should we be listing new output variables in the output rules doc?

**Comment by Myoldmopar** [comment]
No, I don't think that's necessary.

**Comment by mjwitte** [comment]
Tested this, fixed the variable name in the SetupOutputVariable call, output values look reasonable. Will merge after CI comes back clean.

**Comment by eringold** [comment]
@mjwitte @RKStrand Just FYI, the Construction:InternalSource section in the I/O reference is missing all output descriptions and some of the inputs (instead there are a bunch of HAMT construction outputs). I noted this in the big documentation issue here: https://github.com/NREL/EnergyPlus/issues/5603#issuecomment-226952335, but it may have flown under the radar.
If it wasn't and a fix is in the works, please disregard!

**Comment by mjwitte** [comment]
@eringold Thanks for pointing that out.  The new output variable here was simply added to the existing list in the general surface output section.  Will attempt to clean this up here.

**Comment by mjwitte** [comment]
@eringold The HAMT and Construction:InternalSource output sections have been cleaned up (duplicates gone, internal source outputs moved from prior location, outdated material deleted).  Please review [here](http://energyplus.s3-website-us-east-1.amazonaws.com/assets/2016-08-31-EnergyPlus-ee02cc0bef-x86_64-gcc-4.8/) and let me know if it looks ok.

**Comment by mjwitte** [comment]
@RKStrand This is impacting [performance](https://nrel.github.io/EnergyPlusBuildResults/EnergyPlus-ee02cc0befbab1dc6207e8dda177184f9d5750b3-x86_64-Linux-Ubuntu-14.04-gcc-4.8.html#performance) with a 0.4% increase in run time.  Small, but not insignificant.  I see some places where operations related to source temperature/flux/histories are inside an if construct.SourceSinkPresent but not everywhere.  Seems we could avoid these arrays (new and old source variables) entirely for runs without any sources by adding a few ifs and may be a new global AnySourceSinkPresent flag.

**Comment by eringold** [comment]
@mjwitte looks great, thank you!

**Comment by RKStrand** [comment]
@mjwitte I'll look at it and see if I can construct some if-thens to improve things.  I don't want the performance hit unless we absolutely cannot avoid it.  As you mentioned, 0.4% is not huge but to high for such an issue.  If you need to switch this back to unfinished in Pivotal Tracker and reassign to me, that is fine.

File: src/EnergyPlus/HeatBalanceSurfaceManager.cc [hunk 1]
```diff
@@ -3978,13 +4018,17 @@ namespace HeatBalanceSurfaceManager {
...
-			Real64 const QsrcHist1( QsrcHist( SurfNum, 1 ) );
-
 			// Set current inside flux:
-			Real64 const QH_12 = QH[ l21 ] = TH[ l11 ] * construct.CTFCross( 0 ) - TempSurfIn( SurfNum ) * construct.CTFInside( 0 ) + QsrcHist1 * construct.CTFSourceIn( 0 ) + CTFConstInPart( SurfNum ); // Heat source/sink term for radiant systems
+			Real64 const QH_12 = QH[ l21 ] = TH[ l11 ] * construct.CTFCross( 0 ) - TempSurfIn( SurfNum ) * construct.CTFInside( 0 ) + CTFConstInPart( SurfNum ); // Heat source/sink term for radiant systems
 			if ( surface.Class == SurfaceClass_Floor || surface.Class == SurfaceClass_Wall || surface.Class == SurfaceClass_IntMass || surface.Class == SurfaceClass_Roof || surface.Class == SurfaceClass_Door ) {
-				OpaqSurfInsFaceConduction( SurfNum ) = surface.Area * QH_12;
-				OpaqSurfInsFaceConductionFlux( SurfNum ) = QH_12; //CR 8901
+				if ( construct.SourceSinkPresent ) {
+					Real64 const QH_12s = QH[ l21 ] = QH_12 + QsrcHist( SurfNum, 1 ) * construct.CTFSourceIn( 0 );
```
Comment (mjwitte) [inline src/EnergyPlus/HeatBalanceSurfaceManager.cc:4025]: @RKStrand I'm guessing that the small diffs for most files with internal source are coming from here.  Not a problem, really, just a place where the sequence of calculations changed slightly.

**Comment by mjwitte** [comment]
@RKStrand Nice!  Now we have a 0.28% _positive_ speed improvement.

**Comment by RKStrand** [comment]
Whoo-hoo!  Do I get extra credit for such a speed improvement?

**Comment by RKStrand** [comment]
@mjwitte It looks like the pull request is still active.  I have marked Pivotal Tracker back to finished/delivered and switched the owner back to you.  Am I correct in assuming this pull request is still active and that you will finish the review?

**Comment by mjwitte** [comment]
Yes, sir.

**Pull request merged**